File: app/src/main/java/me/destro/android/gitfav/paging/StarredRepositoryDiffCallback.java
package me.destro.android.gitfav.paging;

import androidx.recyclerview.widget.DiffUtil;

import me.destro.android.libraries.github.model.StarredRepository;

public class StarredRepositoryDiffCallback extends DiffUtil.ItemCallback<StarredRepository> {
    @Override
    public boolean areItemsTheSame(StarredRepository oldItem, StarredRepository newItem) {
        return oldItem.id == newItem.id;
    }

    @Override
    public boolean areContentsTheSame(StarredRepository oldItem, StarredRepository newItem) {
        return oldItem.name == newItem.name &&
                oldItem.fullName == newItem.fullName &&
                oldItem.description == newItem.description &&
                oldItem.htmlUrl == newItem.htmlUrl &&
                oldItem.topics == newItem.topics;
    }
}

File: app/src/main/java/me/destro/android/gitfav/features/listing/paging/StarredRepositoryDiffCallback.kt
package me.destro.android.gitfav.features.listing.paging

import androidx.recyclerview.widget.DiffUtil
import me.destro.android.gitfav.domain.model.Repository


class StarredRepositoryDiffCallback : DiffUtil.ItemCallback<Repository>() {
    override fun areItemsTheSame(oldItem: Repository, newItem: Repository): Boolean {
        return newItem.name == oldItem.name
                && newItem.owner == newItem.owner
    }

    override fun areContentsTheSame(oldItem: Repository, newItem: Repository): Boolean {
        return oldItem.name == newItem.name &&
                oldItem.owner == newItem.owner &&
                oldItem.description == newItem.description &&
                oldItem.topics.contentEquals(newItem.topics)
    }
}
